Update regions and hosting suggestions

KevinBanana opened the pull request:
Based on [this discussion](https://discord.com/channels/689979228841836632/692200535348215839/1169011837501706364), AWS is no longer an option so I have updated the hosting suggestions to mention GCP instead.

I also included a bit about GCP in the hosting suggestions. I'm not super experienced with GCP and there are some subjective opinions in there, so extra scrutiny there would be appreciated.

KevinBanana committed:
Update regions and hosting suggestions

KevinBanana committed:
remove aws tip

bvanvugt commented:
Heck yeah! Thanks for writing this up.

Can you also remove the DigitalOcean engine regions from the list? They're also no longer available. Otherwise looks great!

KevinBanana committed:
remove digital ocean supported engine regions

also deleted a snipper about IPv6 support that was only applicable to AWS and DigitalOcean

bvanvugt commented:
Should be live in a couple minutes -- thanks @KevinBanana! :)
